docs/bsp.md

```diff
@@ -159,6 +163,13 @@ trait BuildTarget {
   def data: Option[Json] // Note, matches `any` in the LSP.
 }
 
+object BuildTargetKind {
+  final val Library = 1
+  final val Test = 2
+  final val IntegrationTest = 3
+  final val Bench = 4
```

Comment:
If we keep kind as a single element, 

- it should be a string instead of int, similarly as language IDs are strings
- we are missing `App` and in the future more categories may come up like `iOS`, `Android`

I personally lean towards turning `kind: Int` into `labels: List[String]` where we provide a builtin set of strings like we do with languages.

Reply:
After more thinking and offline discussions, I'm in favor of keeping `BuildTarget.kind: Int` but we should add `final val App = ...`